zipkin/src/main/java/zipkin/internal/JsonCodec.java

```diff
@@ -235,14 +236,14 @@ public BinaryAnnotation fromJson(JsonReader reader) throws IOException {
 
     @Override public int sizeInBytes(BinaryAnnotation value) {
       int sizeInBytes = 0;
-      sizeInBytes += asciiSizeInBytes("{\"key\":\"") + utf8SizeInBytes(value.key);
+      sizeInBytes += asciiSizeInBytes("{\"key\":\"") + jsonEscapedSizeInBytes(value.key);
       sizeInBytes += asciiSizeInBytes("\",\"value\":");
       switch (value.type) {
         case BOOL:
           sizeInBytes += asciiSizeInBytes(value.value[0] == 1 ? "true" : "false");
           break;
         case STRING:
-          sizeInBytes += value.value.length +2; //for quotes
+          sizeInBytes += jsonEscapedSizeInBytes(value.value) + 2; //for quotes
```

Comment:
this was the bugger. the raised issue had special characters in a binary annotation! for json encoding, we need to peek at it.